stdlib/public/core/HashedCollections.swift.gyb

```diff
@@ -2710,33 +2714,30 @@ final internal class _Native${Self}StorageOwner<${TypeParametersDecl}>
         // do nothing, both are null
       } else {
         // keys null, objects nonnull
-        while position < capacity {
+        for position in 0..<count {
           if bridgedNativeStorage.isInitializedEntry(at: position) {
-            unmanagedObjects[i] = bridgedNativeStorage.value(at: position)
+            unmanagedObjects![i] = bridgedNativeStorage.value(at: position)
             i += 1
           }
-          position += 1
         }
       }
     } else {
       if objects == nil {
         // keys nonnull, objects null
-        while position < capacity {
+        for position in 0..<count {
           if bridgedNativeStorage.isInitializedEntry(at: position) {
-            unmanagedKeys[i] = bridgedNativeStorage.key(at: position)
+            unmanagedKeys![i] = bridgedNativeStorage.key(at: position)
             i += 1
           }
-          position += 1
         }
       } else {
         // keys nonnull, objects nonnull
-        while position < capacity {
+        for position in 0..<count {
           if bridgedNativeStorage.isInitializedEntry(at: position) {
-            unmanagedObjects[i] = bridgedNativeStorage.value(at: position)
-            unmanagedKeys[i] = bridgedNativeStorage.key(at: position)
+            unmanagedObjects![i] = bridgedNativeStorage.value(at: position)
+            unmanagedKeys![i] = bridgedNativeStorage.key(at: position)
```

Comment:
Do you really want to check every time around the loop?  Maybe hoist the check out?